download/tasks.py

```diff
@@ -20,25 +21,36 @@ def download_bano(ctx, bano_url, output_file):
 
 
 @task
-def download_oa(ctx, oa_files):
-    temp_dir = tempfile.mkdtemp()
+def download_oa(ctx, oa_filter):
+    tmp_dir = tempfile.mkdtemp()
 
-    oa_files = oa_files.split(",")
-    oa_temp_dir = path.join(temp_dir, "oa")
+    tmp_file = path.join(tmp_dir, "oa.zip")
+    ctx.run(f"mkdir -p {path.dirname(tmp_file)}")
+    ctx.run(f"wget --progress=dot:giga {ctx.oa_url} -O {tmp_file}")
 
-    for oa_filename in oa_files:
-        source_url = f"{ctx.oa_base_url}{oa_filename}.zip"
-        tmp_file = path.join(temp_dir, f"{oa_filename}.zip")
-        ctx.run(f"mkdir -p {path.dirname(tmp_file)}")
-        ctx.run(f"wget --progress=dot:giga {source_url} -O {tmp_file}")
-        ctx.run(f"unzip -o -qq -d {oa_temp_dir} {tmp_file}")
-        ctx.run(f"rm {tmp_file}")
+    oa_tmp_dir = path.join(tmp_dir, "oa")
+    ctx.run(f"unzip -o -qq -d {oa_tmp_dir} {tmp_file}")
+    ctx.run(f"rm {tmp_file}")
 
+    # Collect the list of files to include in the output
+    keep_patterns = (fnmatch.translate(pat) for pat in oa_filter.split(","))
+    pattern = re.compile("|".join(keep_patterns))
+    included_files = []
+
+    for dirname, _, files in walk(oa_tmp_dir):
+        for filename in files:
+            full_path = path.join(dirname, filename)
+            relt_path = full_path.lstrip(oa_tmp_dir)
```

Comment:
`.lstrip()` takes a set of chars as its argument, so is actually not safe in this context.
```suggestion
            relt_path = os.path.relpath(full_path, oa_tmp_dir)
```